GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/ClassificationPanelViewModel.cs

```diff
@@ -0,0 +1,174 @@
+            }
+            set
+            {
+                _currentAnnotation = value;
+                OnPropertyRaised("CurrentAnnotation");
+            }
+        }
+
+        private string _subjectImageSource;
+        public string SubjectImageSource
+        {
+            get
+            {
+                return _subjectImageSource;
+            }
+            set
+            {
+                _subjectImageSource = value;
+                OnPropertyRaised("SubjectImageSource");
+            }
+        }
+
+        public ICommand SelectAnswer { get; set; }
+        public ICommand SubmitClassification { get; set; }
+
+        private AnswerButton _selectedItem;
+        public AnswerButton SelectedItem
+        {
+            get
+            {
+                return _selectedItem;
+            }
+            set
+            {
+                _selectedItem = value;
+                OnPropertyRaised("SelectedItem");
+            }
+        }
+
+        public ClassificationPanelViewModel(Workflow workflow)
+        {
+            GetSubject();
+            LoadCommands();
+            Workflow = workflow;
+            CurrentTask = workflow.Tasks[workflow.FirstTask];
+            CurrentTaskIndex = workflow.FirstTask;
+
+            if (CurrentTask.Answers != null)
+            {
+                CurrentAnswers = ParseTaskAnswers(CurrentTask.Answers);
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyRaised(string propertyname)
+        {
+            PropertyChanged(this, new PropertyChangedEventArgs(propertyname));
+        }
+
+        private void LoadCommands()
+        {
+            SelectAnswer = new CustomCommand(ChooseAnswer, CanSelectAnswer);
+            SubmitClassification = new CustomCommand(SendClassification, CanSendClassification);
+        }
+
+        private async void SendClassification(object obj)
+        {
+            CurrentClassification.Metadata.FinishedAt = DateTime.Now.ToString();
+            CurrentClassification.Annotations.Add(CurrentAnnotation);
+            ApiClient client = new ApiClient();
+            await client.Classifications.Create(CurrentClassification);
+            GetSubject();
+        }
+
+        private bool CanSendClassification(object obj)
+        {
+            return CurrentAnnotation != null;
+        }
+
+        private void ChooseAnswer(object obj)
+        {
+            AnswerButton button = (AnswerButton)obj;
+            SelectedItem = button;
+            CurrentAnnotation = new Annotation(CurrentTaskIndex, button.Index);
+        }
+
+        private bool CanSelectAnswer(object obj)
+        {
+            return CurrentAnswers.Count > 0;
+        }
+
+        public List<AnswerButton> ParseTaskAnswers(List<TaskAnswer> answers)
+        {
+            List<AnswerButton> renderedAnswers = new List<AnswerButton>();
+
+            for (int i = 0; i < answers.Count; i++)
```

Comment:
Can you use a more descriptive name than `i` here?

Reply:
I can. `i` is often used as shorthand for `index`, but it can easily be changed to `index`.

Reply:
Haha, sorry I'm so ticky tacky about names. You can leave it as `i`, I just think the code becomes a little more readable when variable names are more verbose. Even something like `answerIndex` is so much more readable to me than `i`